misk/src/main/kotlin/misk/config/ConfigModule.kt

```diff
@@ -18,11 +23,25 @@ class ConfigModule<T : Config>(
   private val appName: String,
   private val config: T
 ) : KAbstractModule() {
+
   @Suppress("UNCHECKED_CAST")
   override fun configure() {
     bind<String>().annotatedWith<AppName>().toInstance(appName)
     bind(configClass).toInstance(config)
     bindChildConfigClass(configClass)
+
+    val configLoaderProvider = object : Provider<SecretConfigLoader> {
+      @Inject lateinit var resourceLoader: ResourceLoader
+      @Inject lateinit var moshi: Moshi
+      override fun get(): SecretConfigLoader =
+          SecretConfigLoader(resourceLoader, moshi, configClass, config)
+    }
+
+    multibind<Service>().toProvider(Provider<SecretConfigService> {
```

Comment:
It's a little strange for config to register a service - can we just do the loading in-line here, or on-demand when someone calls `value`